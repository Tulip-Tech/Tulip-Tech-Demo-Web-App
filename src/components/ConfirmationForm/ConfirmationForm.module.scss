@use 'src/styles/variables';
@use 'src/styles/theme';
.title {
  margin: 24px 0;
  font-weight: var(--body-font-weight-bold);
  font-size: 26px;
}

.text {
  margin-bottom: 24px;
  font-size: 16px;
}

.button {
  margin: 16px 0;
}

.notSure {
  font-size: 16px;
}

.link {
  margin-bottom: 24px;
  margin-left: 6px;
  color: var(--primary-color);
  font-weight: var(--body-font-weight-bold);
  cursor: pointer;
  &:hover {
    text-decoration: underline;
  }
}
