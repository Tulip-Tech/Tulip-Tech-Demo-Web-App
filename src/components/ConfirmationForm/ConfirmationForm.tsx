import React from 'react';
import { useTranslation } from 'react-i18next';
import { Link, useLocation } from 'react-router-dom';

import styles from './ConfirmationForm.module.scss';

import Button from '#components/Button/Button';
import { addQueryParam } from '#src/utils/location';

type Props = {
  email?: string;
  loggedIn?: boolean;
  onBackToLogin: () => void;
};

const ConfirmationForm: React.FC<Props> = ({ email, onBackToLogin, loggedIn }: Props) => {
  const { t } = useTranslation('account');
  const location = useLocation();

  return (
    <div className={styles.forgotPasswordForm}>
      <h2 className={styles.title}>{t('reset.link_sent')}</h2>
      <p className={styles.text}>{t('reset.link_sent_text', { email: email })}</p>
      <Button onClick={onBackToLogin} className={styles.button} fullWidth color="primary" label={t('reset.back_login')} />
      {!loggedIn && (
        <React.Fragment>
          <span className={styles.notSure}>{t('reset.not_sure')}</span>
          <Link className={styles.link} to={addQueryParam(location, 'u', 'forgot-password')}>
            {t('reset.try_again')}
          </Link>
        </React.Fragment>
      )}
    </div>
  );
};

export default ConfirmationForm;
