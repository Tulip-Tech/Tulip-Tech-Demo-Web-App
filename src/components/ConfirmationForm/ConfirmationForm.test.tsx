import React from 'react';

import ConfirmationForm from './ConfirmationForm';

import { renderWithRouter } from '#test/testUtils';

describe('<ConfirmationForm>', () => {
  test('renders and matches snapshot', () => {
    const { container } = renderWithRouter(<ConfirmationForm onBackToLogin={vi.fn()} />);

    expect(container).toMatchSnapshot();
  });
});
