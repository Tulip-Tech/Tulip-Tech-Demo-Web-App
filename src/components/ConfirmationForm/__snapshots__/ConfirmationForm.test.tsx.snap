// Vitest Snapshot v1

exports[`<ConfirmationForm> > renders and matches snapshot 1`] = `
<div>
  <div>
    <h2
      class="_title_64b7a0"
    >
      reset.link_sent
    </h2>
    <p
      class="_text_64b7a0"
    >
      reset.link_sent_text
    </p>
    <button
      class="_button_f8f296 _button_64b7a0 _primary_f8f296 _outlined_f8f296 _fullWidth_f8f296"
    >
      <span>
        reset.back_login
      </span>
    </button>
    <span
      class="_notSure_64b7a0"
    >
      reset.not_sure
    </span>
    <a
      class="_link_64b7a0"
      href="/?u=forgot-password"
    >
      reset.try_again
    </a>
  </div>
</div>
`;
